storage/modules/balance_storage.go

```diff
@@ -1097,7 +1097,16 @@ func (b *BalanceStorage) BootstrapBalances(
 	dbTransaction := b.db.Transaction(ctx)
 	defer dbTransaction.Discard(ctx)
 
-	for _, balance := range balances {
+	for i, balance := range balances {
+		// Commit transaction batch by batch rather than commit at one time.
+		// This helps reduce memory usage and improve running time when bootstrap_balances.json
+		// contains huge number of accounts.
+		if i != 0 && i%utils.MaxEntrySizePerTxn == 0 {
```

Comment:
What about the last batch that is less than maxEntry?
ex: you have a list of 700, and maxEntry is 600.
Will we commit the [601, 700]?

Reply:
https://github.com/coinbase/rosetta-sdk-go/pull/367/files#diff-66c37bf21a824a6a6715142e0586224424d4c6cea795913fa549112ca51edc42R1142 - this line will commit the last tx. I need to see how I can write a unit test with smaller number to test it automatically but I tested it locally with maxEntry 2 with various cases.